test/WireMock.Net.Tests/FluentAssertions/WireMockAssertionsTests.cs

```diff
@@ -740,6 +744,41 @@ public async Task HaveReceived1Call_WithBodyAsJson()
         server.Stop();
     }
 
+    [Fact]
+    public async Task WithBodyAsJson_When_NoMatch_ShouldHaveCorrectErrorMessage()
+    {
+        // Arrange
+        var server = WireMockServer.Start();
+
+        server
+            .Given(Request.Create().WithPath("/a").UsingPost().WithBodyAsJson(new { x = "y" }))
+            .RespondWith(Response.Create().WithBody("A response"));
+
+        // Act
+        var httpClient = new HttpClient();
+
+        var requestBody = new
+        {
+            x = "123"
+        };
+        await httpClient.PostAsJsonAsync($"{server.Url}/a", requestBody);
+
+        // Assert
+        Action act = () => server
+            .Should()
+            .HaveReceived(1)
+            .Calls()
+            .WithBodyAsJson(new { x = "y" })
+            .And
+            .UsingPost();
+
+        act.Should()
+            .Throw<Exception>()
+            .WithMessage("Expected wiremockserver to have been called using body \"{\"x\":\"y\"}\", but didn't find it among the body/bodies \"{\"x\":\"123\"}\".");
```

Comment:
As the `LangVersion` is 11, you could use a raw string literal `"""` to avoid escaping the quotes. It would probably be more readable